File: NaoBlocks.Core/Models/UserSettings.cs
using Newtonsoft.Json;

namespace NaoBlocks.Core.Models
{
    public class UserSettings
    {
        public int AllocationMode { get; set; }

        public bool Conditionals { get; set; }

        public int? CurrentExercise { get; set; }

        public string? CurrentTutorial { get; set; }

        public bool Dances { get; set; }

        public bool Events { get; set; }

        public bool Loops { get; set; }

        public string? RobotType { get; set; }
        
        public string? RobotTypeId { get; set; }

        public string? RobotId { get; set; }

        public bool Sensors { get; set; }

        public bool Simple { get; set; }

        public bool Tutorials { get; set; }

        public bool Variables { get; set; }
    }
}

File: Clients/Angular/src/app/data/user-settings.ts
export class UserSettings {
    simple: boolean = false;
    events: boolean = false;

    conditionals: boolean = false;
    dances: boolean = false;
    loops: boolean = false;
    sensors: boolean = false;
    tutorials: boolean = false;
    variables: boolean = false;

    robotType?: string;
    robotId?: string;
    allocationMode?: number;

    customBlockSet?: string;
}
